witness_beacon.go

```diff
@@ -129,5 +130,16 @@ func (p *preimageBeacon) AddPreimage(pre []byte) error {
 	return nil
 }
 
+// SettleInvoice attempts to settle an existing invoice with the given payment
+// hash. ErrInvoiceNotFound is returned if an invoice is not found.
+func (p *preimageBeacon) SettleInvoice(paymentHash [32]byte,
+	amt lnwire.MilliSatoshi) error {
+
+	p.Lock()
+	defer p.Unlock()
+
+	return p.invoices.SettleInvoice(paymentHash, amt)
```

Comment:
It is currently just a pass-through to InvoiceRegistry, so maybe it is better to make InvoiceRegistry a dependency of ChainArbitratorConfig so that the resolver can access it.

Reply:
Fixed - this made the changes even simpler.